core/src/auth_service/db_handler.bal

```diff
@@ -0,0 +1,85 @@
+import ballerina/config as conf;
+import ballerina/mongodb;
+
+// Mongodb configurations.
+mongodb:ClientConfig mongoConfig = {
+    host: conf:getAsString("DB_HOST"),
+    username: conf:getAsString("DB_USER_NAME"),
+    password: conf:getAsString("DB_PASSWORD"),
+    options: {sslEnabled: false, serverSelectionTimeout: 500}
+};
+mongodb:Client mongoClient = check new (mongoConfig);
+mongodb:Database mongoDatabase = check mongoClient->getDatabase("EnvironmentPlatform");
+mongodb:Collection usersCollection = check mongoDatabase->getCollection("users");
+mongodb:Collection ministryCollection = check mongoDatabase->getCollection("ministries");
+mongodb:Collection adminCollection = check mongoDatabase->getCollection("admins");
+
+# The `getUser` function will authenticate the user with the given email and the hashed password.
+# 
+# + email - Email of the user.
+# + password - Password of the user.
+# + return - This function will return either the user information as json or an appropriate error.
+function getUser(string email, string password) returns json|error {
+    map<json>[] users = check usersCollection->find({email: email, password: password});
+    map<json>[] find = check usersCollection->find({email: email});
+    if (find.length() > 1) {
+        return error("Multiple users", message = "There are multiple users with same email:" + email + ".");
```

Comment:
We shouldn't let this happen in the first place right? There can't be multiple users with the same email. I suggest adding a flow to check for existing records when adding a new user if not already added.

Reply:
Yeah it must be done when user creation.

Reply:
I just added this for completion.

Reply:
If it is done in user creation it saves us from having to check for multiple users in all other places. So once that validation is added to the user creation we can remove these checks.

Reply:
I will remove it then.